helm-chart/binderhub/values.yaml

```diff
@@ -145,6 +145,18 @@ jupyterhub:
       enabled: false
 
 deployment:
+  readinessProbe:
+    enabled: true
+    initialDelaySeconds: 0
+    periodSeconds: 5
+    failureThreshold: 1000  # we rely on the liveness probe to resolve issues if needed
```

Comment:
For my education: what does the comment mean/try to say? Why not use a startup probe which seems to be made for exactly our case of "pod can take time to start up"?

I think setting the failure threshold to 1000 means that the readiness probe will never(*) become "not ready" again after it becomes ready for the first time.

Why not use an initial delay similar to the liveness probe to cover the expected time the process needs to start up (I think the initial delay only starts counting after the container image has been pulled but couldn't confirm that).

(*) in reality it would become "not ready" if the probe fails for ~5000s, so about 1.3hours

Reply:
__readinessProbe__
- Required to make a pod Ready.
- A failure of this -> the container is becoming Unready
- Ready is required for the pod to receive traffic from k8s Services
- Ready is required for `kubectl rollout status --wait` checks of a deployment resource.
- A readinessProbe with a very long failureThreshold makes it act as a delay to become Ready during startup, but will then always remain Ready.
- To let a readinessProbe fail following having become ready, the purpose would be to temporarily remove it from receiving traffic from k8s Services, perhaps to recover or similar, but this is only useful in specific situations I don't think is applicable to our Binder pod.

__livenessProbe__
- A failure of this -> the container is getting restarted

__startupProbe__
- k8s 1.16+ required
- doesn't influence the pod's Ready status
- serves the purpose only to complement the livenessProbe for slow startups, see [this docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes)

__Meaning of comment?__
I meant to convey that we rely on the livenessProbe's functionality to recover from bad states, not by simply making the pod Unready which can be unhelpful.

__Reference__
startupProbe and such is discussed also in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1732#issuecomment-704288546